corehq/apps/users/views/__init__.py

```diff
@@ -869,6 +869,7 @@ def _update_role_from_view(domain, role_data):
 
     assignable_by = role_data["assignable_by"]
     role.set_assignable_by_couch(assignable_by)
+    role._migration_do_sync()  # update permissions and assignable_by
```

Comment:
Is the reason this did not happen automatically during operations that persist changes to SQL because `RoleAssignableBy` is a related model of `SQLUserRole`?

Is it important that `set_assignable_by_couch` does not call this when it is called in other places? If not, would it make sense to call it in that method or maybe in `set_assignable_by`?

Reply:
We could do that but then in a method like this we'd be syncing to couch 3 times, once in `save` once in `set_assignable_by` and once in `set_permissions`. I specifically did not do that for the purposes of the migration but these changes are low volume enough that it's likely not an issue now.

I'll PR that change and link it here.